google/cloud/storage/examples/storage_bucket_samples.cc

```diff
@@ -221,6 +221,87 @@ void PatchBucketStorageClassWithBuilder(google::cloud::storage::Client client,
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  auto label_key = ConsumeArg(argc, argv);
+  auto label_value = ConsumeArg(argc, argv);
+  //! [add bucket label] [START storage_add_bucket_label]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name, std::string label_key,
+     std::string label_value) {
+    gcs::BucketMetadata updated_metadata = client.PatchBucket(
+        bucket_name,
+        gcs::BucketMetadataPatchBuilder().SetLabel(label_key, label_value));
+    std::cout << "Successfully set label " << label_key << " to " << label_value
+              << " on bucket  " << bucket_name << ".";
+    std::cout << " The bucket labels are now:";
+    for (auto const& kv : updated_metadata.labels()) {
+      std::cout << "\n  " << kv.first << ": " << kv.second;
+    }
+    std::cout << std::endl;
+  }
+  //! [add bucket label] [END storage_add_bucket_label]
+  (std::move(client), bucket_name, label_key, label_value);
+}
+
+void GetBucketLabels(google::cloud::storage::Client client, int& argc,
+                     char* argv[]) {
+  if (argc != 2) {
+    throw Usage{"get-bucket-labels <bucket-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [get bucket labels] [START storage_get_bucket_labels]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name) {
+    gcs::BucketMetadata metadata =
+        client.GetBucketMetadata(bucket_name, gcs::Fields("labels"));
+    if (metadata.label_count() == 0U) {
+      std::cout << "The bucket " << bucket_name << " has no labels set."
+                << std::endl;
+      return;
+    }
+    std::cout << "The labels for bucket " << bucket_name << " are:";
+    for (auto const& kv : metadata.labels()) {
+      std::cout << "\n  " << kv.first << ": " << kv.second;
+    }
+    std::cout << std::endl;
+  }
+  //! [get bucket label] [END storage_get_bucket_labels]
+  (std::move(client), bucket_name);
+}
+
+void RemoveBucketLabel(google::cloud::storage::Client client, int& argc,
+                       char* argv[]) {
+  if (argc != 3) {
+    throw Usage{"remove-bucket-label <bucket-name> <label-key>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  auto label_key = ConsumeArg(argc, argv);
+  //! [remove bucket label] [START storage_remove_bucket_label]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name, std::string label_key) {
+    gcs::BucketMetadata updated_metadata = client.PatchBucket(
+        bucket_name, gcs::BucketMetadataPatchBuilder().ResetLabel(label_key));
+    std::cout << "Successfully reset label " << label_key << " on bucket  "
+              << bucket_name << ".";
+    if (updated_metadata.label_count() == 0U) {
+      std::cout << " The bucket now has no labels." << std::endl;
+      return;
+    }
+    std::cout << " The bucket labels are now:";
+    for (auto const& kv : updated_metadata.labels()) {
```

Comment:
use type instead of auto.

Reply:
Not in this case I think.  It is idiomatic to iterate using `auto` (some folks recommend `auto&&`, but I digress.  They pretty much invented `auto` so we wouldn't have to type `std::map<std::string,std::string>::value_type const&` in places like this, or worse `std::pair<std::string const, std::string> const&`

Reply:
Ah, right. My auto -> type instincts are too strong. Please disregard.